tools/README.md

```diff
@@ -0,0 +1,31 @@
+# Monorepo Infrastructure & Tools
+
+This document aims to provide an outline for the monorepo's infrastructure as well as the rationale behind the decisions we've made.
+
+## Task Orchestration
+
+Our repository makes aggressive use of [parallelization using PNPM's `--filter` syntax](https://pnpm.io/filtering). This allows us to minimize the amount of time developers spend waiting for tasks like builds to complete. Each project within the monorepo will contain the following scripts if applicable:
+
+```json
+{
+	"scripts": {
+		"build": "pnpm --if-present --workspace-concurrency=Infinity --stream --filter=\"$npm_package_name...\" '/^build:project:.*$/'",
+		"lint": "pnpm --if-present '/^lint:lang:.*$/'",
+		"lint:fix": "pnpm --if-present '/^lint:fix:lang:.*$/'",
+		"watch:build": "pnpm --if-present --workspace-concurrency=Infinity --filter=\"$npm_package_name...\" --parallel '/^watch:build:project:.*$/'",
+	}
+}
+```
+
+These scripts outline the naming scheme used in order to facilitate [task parallelization using regular expressions](https://pnpm.io/cli/run#running-multiple-scripts). **To ensure consistency across the monorepo, these scripts should not be edited.** New scripts should be added using the naming scheme outlined by the above regular expressions, for example, `build:project:bundle` might be a script to run a tool like `webpack`. We also utilize a number of PNPM options to ensure a positive developer experience:
+
+- `--if-present`: Ensures that PNPM will not error if a script is not found.
+- `--workspace-concurrency=Infinity`: Runs as many of the tasks in parallel as possible.
+- `--stream`: Makes the script output legible by putting all of their output into a single stream.
+- `--filter="$npm_package_name..."`: This filter tells PNPM that we want to run the script against the current project _and_ all of its dependencies down the graph.
```

Comment:
Maybe we should tag a quick example here?

For example:
Lets take the `currency` package, which has the `number` and `eslint-plugin` as dependencies and dev dependencies.
When running `pnpm --filter=\"$npm_package_name...\" '/^build:project:.*$/'"`, it will run the `build:project:*` commands first on the `number` and `eslint-plugin` packages and than on the `currency` package.
`pnpm` will know this as the dependency use the `workspace:*` as version, meaning they exist within the workspace/monorepo.

Reply:
I like the idea, but we need to reframe it a bit. If we provide this detailed example, it might become outdated, and we will likely not notice that it is outdated. Still, here my attempt to bring your point into the documentation: 5a2977db3f86d2cc85d1d90c2e4108aef8c5e1ea